esp/src/src-react/routes.tsx

```diff
@@ -130,19 +131,40 @@ export const routes: RoutesEx = [
                 })
             },
             {
-                path: "/:Name", action: (ctx, params) => import("./components/FileDetails").then(_ => {
-                    return <_.FileDetails cluster={undefined} logicalFile={params.Name as string} />;
-                })
+                path: "/:Name", action: (ctx, params) => {
+                    const FileContextProvider = React.lazy(() => import("./components/contexts/FileContext"));
```

Comment:
This is a bit different, but I wasn't quite sure what would be better for conditionally loading more than one component in these router callbacks. I found this React.lazy in the docs, and it required wrapping in this React.Suspense to render properly. lazy() also expects the component being loaded to have a default export, hence the one line change to FileDetails.